osdconfig/proto/config.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+// this package defines the configuration parameters and interfaces
+// requires to run portwork services
+package proto;
+
+service Spec {
+    rpc GetGlobalSpec(Empty) returns (GlobalConfig) {}
+    rpc SetGlobalSpec(GlobalConfig) returns (Ack) {}
+    rpc GetClusterSpec(Empty) returns (ClusterConfig) {};
+    rpc SetClusterSpec(ClusterConfig) returns (Ack) {};
+    rpc GetNodeSpec(NodeID) returns (NodeConfig) {};
+    rpc SetNodeSpec(NodeConfig) returns (Ack) {};
+}
+
+message Empty {}
+message Ack {
+    int64 n = 1;
+}
+message NodeID {
+    string i_d = 1;
+}
+
+// Global Conf contains all the configuration data
+message GlobalConfig {
+    ClusterConfig cluster_conf = 1;
+    NodesConfig nodes_conf = 2;
+}
+
+message NodesConfig {
+    map<string, NodeConfig> node_conf = 1;
+}
+
+message NodeConfig {
+    string node_id = 1;
```

Comment:
nit: Should this have been node_i_d to be consistent with the golint?
Also, I am not sure how i_d works for other languages, just in case some one decides to have the Spec service implemented in another language.

Reply:
good point. node_i_d will become nodeID in Go. let's make a call on it. i am okay both ways.